vcd/resource_vcd_nsxt_alb_virtual_service.go

```diff
@@ -0,0 +1,346 @@
+	return &schema.Resource{
+		CreateContext: resourceVcdAlbVirtualServiceCreate,
+		ReadContext:   resourceVcdAlbVirtualServiceRead,
+		UpdateContext: resourceVcdAlbVirtualServiceUpdate,
+		DeleteContext: resourceVcdAlbVirtualServiceDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdAlbVirtualServiceImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"org": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Description: "The name of organization to use, optional if defined at provider " +
+					"level. Useful when connected as sysadmin working across different organizations",
+			},
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The name of VDC to use, optional if defined at provider level",
+			},
+			"edge_gateway_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway ID in which ALB Pool should be created",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of ALB Virtual Service",
+			},
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Description of ALB Virtual Service",
+			},
+			"pool_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway ID in which ALB Pool should be created",
+			},
+			"service_engine_group_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway ID in which ALB Pool should be created",
+			},
+			"ca_certificate_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "A set of root certificate IDs to use when validating certificates presented by pool members",
+			},
+			"enabled": &schema.Schema{
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     true,
+				Description: "Virtual Service is enabled or disabled (default true)",
+			},
+			"virtual_ip_address": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Virtual IP address (VIP) for Virtual Service",
+			},
+			"application_profile_type": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "HTTP, HTTPS, L4, L4_TLS",
+			},
+			"service_port": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     nsxtAlbVirtualServicePort,
+			},
+		},
+	}
+}
+
+var nsxtAlbVirtualServicePort = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"start_port": &schema.Schema{
+			Type:        schema.TypeInt,
+			Required:    true,
+			Description: "Starting port in the range",
+		},
+		"end_port": &schema.Schema{
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Description: "Starting port in the range",
```

Comment:
Wrong description.

Reply:
Fixed this and a few more erroneous descriptions. Hopefully it should be good for all.